test/TestImport.hs

```diff
@@ -24,4 +27,28 @@ withApp = before $ do
         ["config/test-settings.yml", "config/settings.yml"]
         []
         ignoreEnv
-    makeFoundation settings
+    foundation <- makeFoundation settings
+    wipeDB foundation
+    return foundation
+
+-- This function will truncate all of the tables in your database.
+-- 'withApp' calls it before each test, creating a clean environment for each
+-- spec to run in.
+wipeDB :: App -> IO ()
+wipeDB app = do
+    runDBWithApp app $ do
+        tables <- getTables
+        let queries = map ("TRUNCATE TABLE " ++ ) tables
+
+        -- In MySQL, a table cannot be truncated if another table references it via foreign key.
+        -- Since we're wiping both the parent and child tables, though, it's safe
+        -- to temporarily disable this check.
+        rawExecute "SET foreign_key_checks = 0;" []
+        forM_ queries (\q -> rawExecute q [])
+        rawExecute "SET foreign_key_checks = 1;" []
+    return ()
+
+getTables :: MonadIO m => ReaderT SqlBackend m [Text]
+getTables = do
+    tables <- rawSql "SHOW TABLES;" []
```

Comment:
It looks like tools like DatabaseCleaner for Ruby just [special case `schema_migrations`](https://github.com/DatabaseCleaner/database_cleaner/blob/a1ac42cf41ea35f2bd88d507e7d2eb323ff1b679/lib/database_cleaner/active_record/deletion.rb#L58). That wouldn't be a huge deal for someone using a schema migration tool to blacklist, though.

I can look into getting the table names from the Persistent DSL if you'd like.